app/src/main/java/sdumchykov/androidApp/presentation/MainActivity.kt

```diff
@@ -1,30 +1,30 @@
 package sdumchykov.androidApp.presentation
 
 import android.os.Bundle
-import androidx.appcompat.app.AppCompatActivity
 import androidx.navigation.NavController
 import androidx.navigation.fragment.NavHostFragment
 import dagger.hilt.android.AndroidEntryPoint
 import sdumchykov.androidApp.R
+import sdumchykov.androidApp.databinding.ActivityMainBinding
+import sdumchykov.androidApp.presentation.base.BaseActivity
 
 @AndroidEntryPoint
-class MainActivity : AppCompatActivity() {
+class MainActivity : BaseActivity<ActivityMainBinding>(ActivityMainBinding::inflate) {
 
     private lateinit var navController: NavController
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
 
-        val navHostFragment = supportFragmentManager
-            .findFragmentById(R.id.mainActivityNavHostFragment) as NavHostFragment
+//        val findFragmentById = supportFragmentManager.findFragmentById(R.id.mainActivityNavHostFragment) // це типу Fragment?
+//        val mainActivityNavHostFragment = binding.mainActivityNavHostFragment // це типу FragmentContainerView
+//        TODO чому треба через supportFragmentManager виконувати наступну стрічку, а через binding.mainActivityNavHostFragment не можна (пише "я ніколи не приведу тип")?
```

Comment:
Translate into English, never write comments in other from English languages even for yourself